luxtronik/datatypes.py

```diff
@@ -846,6 +846,105 @@ def to_heatpump(cls, value):
 
         return val
 
+class HeatPumpState(SelectionBase):
+    """HeatPumpState datatype, converts from and to list of HeatPumpState codes."""
+
+    datatype_class = "selection"
+
+    codes = {
+        0: "Idle",     # Heatpump is idle
+        1: "Running",  # Heatpump is running
+    }
+
+class ModeState(SelectionBase):
+    """ModeState datatype, converts from and to list of ModeState codes."""
+
+    datatype_class = "selection"
+
+    codes = {
+        0: "Disabled",    # Heating / Hot water is disabled
+        1: "No request",  # Heating / Hot water currently not requested
+        2: "Requested",   # Heating / Hot water requested but currently not running
+        3: "Running",     # Heating / Hot water running
+    }
+
+class ControlMode(SelectionBase):
+    """ControlMode datatype, converts from and to list of ControlMode codes."""
+
+    datatype_class = "selection"
+
+    codes = {
+        0: "Off",       # System value is used
+        1: "Setpoint",  # Setpoint register value is used
+        2: "Offset",    # System values + offset register value is used
+        3: "Level",
+    }
+
+
+class LpcMode(SelectionBase):
+    """LpcMode datatype, converts from and to list of LpcMode codes."""
+
+    datatype_class = "selection"
+
+    codes = {
+        0: "No-Limit",
+        1: "Soft-Limit",
+        2: "Hard-Limit",
```

Comment:
Nit-picking: Where are the dashes (`-`) coming from as separator? Are whitespaces allowed here?